test/Interop/Cxx/class/zero-sized-field.swift

```diff
@@ -24,4 +24,18 @@ FieldsTestSuite.test("Zero sized field") {
   expectEqual(s.b.getNum(), 42)
 }
 
+FieldsTestSuite.test("Field padding reused") {
+  var s = ReuseFieldPadding()
+  s.c = 5
+  expectEqual(Int(s.offset()),  MemoryLayout<ReuseFieldPadding>.offset(of: \.c)!)
+  expectEqual(s.c, 5)
+  expectEqual(s.get_c(), 5)
+  s.set_c(6)
+  expectEqual(s.c, 6)
+  expectEqual(s.get_c(), 6)
+  var s2 = s
+  expectEqual(s2.c, 6)
+  expectEqual(s2.get_c(), 6)
```

Comment:
What happens when you have two instances with different values of the opaque field, and you try to compare them for equality?

Reply:
See my answer above, we cannot automatically conform this type to Equatable.